### PR TITLE
Py36 json support added

### LINKED ISSUE #2243: Take advantage of Py3.6 json support for bytes
[Python 3.6 added support for loading json from str or bytes.](https://docs.python.org/3.6/library/json.html#json.loads) It handles detection of the three encodings supported by JSON. Would be nice to take advantage of this if the user is using 3.6+ instead of our basic "assume it's UTF-8" approach.

Would have to make sure simplejson also allows this (I think it does).

Then again, I'm not sure there have been any reports of Flask decoding JSON incorrectly. Maybe no real clients are encoding with anything besides UTF-8?

### PR DESCRIPTION
**Pull request by akohli96**
Resolves #2243 

1. Added variable in _config.py to check if system is 3.6+
2. Added logic in json.py to check if system is 3.6+ to simply decode json 

EDIT : 3.6 to 3.6+

### CODE REVIEW

File: flask/json.py [hunk 1]
```diff
@@ -154,8 +154,9 @@ def loads(s, **kwargs):
     application on the stack.
     """
     _load_arg_defaults(kwargs)
-    if isinstance(s, bytes):
-        s = s.decode(kwargs.pop('encoding', None) or 'utf-8')
+    if not PY36:
```
Comment (ThiefMaster) [inline flask/json.py]: This doesn't cover 3.7, ...

Comment (ThiefMaster) [inline flask/json.py]: `if sys.version_info[:2] < (3, 6)` would work

Comment (akohli96) [inline flask/json.py]: Great catch @ThiefMaster.
Fixing it right now!

Comment (akohli96) [inline flask/json.py]: I have kept it naming style consistent with PY2.

**Comment by akohli96** [comment]
So, its failing on REQUIREMENTS with no simplejson (devel and release).

**Comment by davidism** [comment]
This no longer does anything about the issue. It only skips decoding for simplejson, while the point was to skip decoding for the built in JSON. It looks like we just can't do this if we want to support non-standard charsets.

**Comment by akohli96** [comment]
[ The input encoding should be UTF-8, UTF-16 or UTF-32](https://docs.python.org/3.6/library/json.html#json.loads) for 3.6 json.loads to work

**Pull request closed without merging**